mm2src/coins/siacoin/sia_withdraw.rs

```diff
@@ -0,0 +1,167 @@
+use crate::siacoin::{siacoin_from_hastings, siacoin_to_hastings, SiaCoin, SiaFeeDetails, SiaFeePolicy,
+                     SiaTransactionTypes};
+use crate::{MarketCoinOps, PrivKeyPolicy, TransactionData, TransactionDetails, TransactionType, WithdrawError,
+            WithdrawRequest, WithdrawResult};
+use common::now_sec;
+use mm2_err_handle::mm_error::MmError;
+use mm2_err_handle::prelude::*;
+use sia_rust::http::endpoints::GetAddressUtxosResponse;
+use sia_rust::spend_policy::SpendPolicy;
+use sia_rust::transaction::{SiacoinOutput, V2TransactionBuilder};
+use sia_rust::types::{Address, Currency};
+use sia_rust::Keypair;
+use std::str::FromStr;
+
+pub struct SiaWithdrawBuilder<'a> {
+    coin: &'a SiaCoin,
+    req: WithdrawRequest,
+    from_address: Address,
+    key_pair: &'a Keypair,
+}
+
+impl<'a> SiaWithdrawBuilder<'a> {
+    #[allow(clippy::result_large_err)]
+    pub fn new(coin: &'a SiaCoin, req: WithdrawRequest) -> Result<Self, MmError<WithdrawError>> {
+        let (key_pair, from_address) = match &coin.0.priv_key_policy {
+            PrivKeyPolicy::Iguana(key_pair) => {
+                let from_address = SpendPolicy::PublicKey(key_pair.public()).address();
+                (key_pair, from_address)
+            },
+            _ => {
+                return Err(WithdrawError::UnsupportedError(
+                    "Only Iguana keypair is supported for Sia coin for now!".to_string(),
+                )
+                .into());
+            },
+        };
+
+        Ok(SiaWithdrawBuilder {
+            coin,
+            req,
+            from_address,
+            key_pair,
+        })
+    }
+
+    #[allow(clippy::result_large_err)]
+    fn select_outputs(
+        &self,
+        mut unspent_outputs: GetAddressUtxosResponse,
+        total_amount: u128,
```

Comment:
Is it sufficient to use u128 for amount?
I took a quick look at sia code: seems, they have huge precision for Currency type (10^24)